algebird-core/src/main/scala/com/twitter/algebird/Aggregator.scala

```diff
@@ -48,6 +48,47 @@ object Aggregator extends java.io.Serializable {
   }
 
   /**
+   * Obtain a [[MonoidAggregator]] that uses an efficient append operation for faster aggregation
+   * @tparam F Data input type
+   * @tparam T Aggregating [[Monoid]] type
+   * @param appnd Function that appends the [[Monoid]].  Defines the [[append]] method for this aggregator.
+   * Analogous to the 'seqop' function in Scala's sequence 'aggregate' method
+   * @param m The [[Monoid]] type class
+   */
+  def appendMonoid[F, T](appnd: (T, F) => T)(implicit m: Monoid[T]): MonoidAggregator[F, T, T] =
+    appendMonoid(appnd, identity[T]_)(m)
+
+  /**
+   * Obtain a [[MonoidAggregator]] that uses an efficient append operation for faster aggregation
+   * @tparam F Data input type
+   * @tparam T Aggregating [[Monoid]] type
+   * @tparam P Presentation (output) type
+   * @param appnd Function that appends the [[Monoid]].  Defines the [[append]] method for this aggregator.
+   * Analogous to the 'seqop' function in Scala's sequence 'aggregate' method
+   * @param pres The presentation function
+   * @param m The [[Monoid]] type class
+   */
+  def appendMonoid[F, T, P](appnd: (T, F) => T, pres: T => P)(implicit m: Monoid[T]): MonoidAggregator[F, T, P] =
```

Comment:
@erikerlandson I think the question is whether the `append()` optimization is more effective than whatever optimization is in `sumOption`. They're both about eliminating unnecessary intermediate values of the type we have a `Monoid` on, but in different ways - it's sorta a question of, for each `reduce(left: T, right: T)`, whether you need the left side to actually be a T (which `sumOption` eliminates) or the right side to actually be a `T` (which `append` eliminates). You could imagine an `appendAll` which requires neither to be, but that feels even more invasive for the `Aggregator` to eliminate vs. the `AppendMonoid` like you proposed.